File: katt3/avian/submissions/accepted/js_python2/avian.py
#!/usr/bin/env python2
def encode(C, K, N, X):
    encoding = [[] for i in range(4)]
    for i in range(N):
        sub = X[i*3:i*3+3]
        encoding[0].append(sub[0])
        encoding[1].append(sub[1])
        encoding[2].append(sub[2])
        encoding[3].append(chr(ord(sub[0]) ^ ord(sub[1]) ^ ord(sub[2])))
    encoding = [''.join(x) for x in encoding]
    return encoding

def decode(C, K, N, Y, I):
    ret = []
    for i in range(N):
        x = 0
        add = ['\0' for _ in range(3)]
        for j in range(3):
            if I[j] < 3:
                add[I[j]] = Y[j][i]
            x ^= ord(Y[j][i])

        for j in range(3):
            if add[j] == '\0':
                add[j] = chr(x)
        ret.append(''.join(add))
    return ''.join(ret)

File: katt3/avian/submissions/accepted/js_cpp/avian.cpp
#include <string>
#include <vector>

#include "avian.h"

using namespace std;

vector<string> encode(int C, int K, int N, string X) {
	vector<string> encoding(4);
	for (int i = 0; i < N; i++) {
		string sub = X.substr(i * 3, 3);
		encoding[0].push_back(sub[0]);
		encoding[1].push_back(sub[1]);
		encoding[2].push_back(sub[2]);
		encoding[3].push_back(sub[0] ^ sub[1] ^ sub[2]);
	}
	return encoding;
}

string decode(int C, int K, int N, vector<string> Y, vector<int> I) {
	string ret;
	for (int i = 0; i < N; i++) {
		char x = 0;
		string add(3, '\0');
		for (int j = 0; j < 3; j++) {
			if (I[j] < 3) {
				add[I[j]] = Y[j][i];
			}
			x ^= Y[j][i];
		}
		for (int j = 0; j < 3; j++) {
			if (add[j] == 0) add[j] = x;
		}
		ret += add;
	}
	return ret;
}
